assets/evaluation_on_cloud/environments/evaluations-built-in/context/evaluate_on_data.py

```diff
@@ -77,12 +101,134 @@ def get_evaluator_config(command_line_args):
+    if data_mapping is None:
+        data_mapping = DEFAULT_DATA_MAPPING
+        logger.info(f"Using default dataMapping: {data_mapping}")
+
+    system_message = target_config.get(SYSTEM_MESSAGE, "")
+    few_shot_examples = target_config.get(FEW_SHOT_EXAMPLES, [])
+
+    logger.info("Creating ModelTarget with values:")
+    logger.info(f"  - Endpoint: {endpoint}")
+    logger.info(f"  - ApiKey: {'[HIDDEN]' if api_key_value else 'MISSING'}")
+    logger.info(f"  - ModelParams: {model_params}")
+    logger.info(f"  - SystemMessage: {system_message}")
+    logger.info(f"  - FewShotExamples: {few_shot_examples}")
+    logger.info(f"  - DataMapping: {data_mapping}")
+
+    model_target = ModelTarget(
+        endpoint=endpoint,
+        api_key=api_key_value,
+        model_params=model_params,
+        system_message=system_message,
+        few_shot_examples=few_shot_examples,
+    )
+
+    return (model_target, data_mapping)
+
+
+def apply_target_on_data(data, model_target, data_mapping):
+    """Apply target on input data."""
+    df = pd.read_json(data, lines=True)
+    mapping = {}
+    for key, value in data_mapping.items():
+        # ignore mappings with empty values
+        if value == "":
+            continue
+        match = re.search(DATA_MAPPING_REGEX_PATTERN, value)
+        if match:
+            mapping[key] = match.group(1)
+        else:
+            logger.info(
+                f'Data mapping did not match the format "{key}":"${{data.<colName in dataset>}}", '
+                f'using given {value}'
+            )
+            mapping[key] = value
+
+    if QUERY_KEY not in mapping:
+        logger.info("Using default mapping for query column")
+        mapping[QUERY_KEY] = QUERY_KEY
+    if CONTEXT_KEY not in mapping:
+        logger.info("Using default mapping for context column")
+        mapping[CONTEXT_KEY] = CONTEXT_KEY
+
+    for index, row_object in df.iterrows():
+        row = row_object.to_dict()
+        mapped_row = {}
+        for actual_col, input_data_col in mapping.items():
+            if input_data_col in row:
+                mapped_row[actual_col] = row[input_data_col]
+            elif actual_col == QUERY_KEY:
+                raise ValueError(
+                    f"'{QUERY_KEY}' is missing in the data_mapping. "
+                    f"The {QUERY_KEY} column mapping is required."
+                )
+
+        query = mapped_row.get(QUERY_KEY, "")
+        context = mapped_row.get(CONTEXT_KEY, "")
+        response = model_target.generate_response(query=query, context=context)
+        df.loc[index, GENERATED_RESPONSE_KEY] = response
+
+    output_dir = data.replace(INPUT_EVAL_DATA_DIR, OUTPUT_EVAL_DATA_DIR)
+    os.makedirs(os.path.dirname(output_dir), exist_ok=True)
+    input_filename = os.path.basename(data)
+    output_filename = f"{os.path.splitext(input_filename)[0]}_output.jsonl"
+    output_file = os.path.join(os.path.dirname(output_dir), output_filename)
```

Comment:
is this output_dir needed because you are first creating the dataset and then passing it in the evaluate_function? So you need the location to read it in it?